index.js

```diff
@@ -56,17 +56,19 @@ exports.lintFiles = (patterns, opts) => {
 	opts = optionsManager.preprocess(opts);
 	patterns = patterns.length === 0 ? ['**/*'] : arrify(patterns);
 
-	const gitIgnores = optionsManager.getGitIgnores(opts);
-	const glob = patterns.concat(gitIgnores);
+	const ignoreFilter = optionsManager.getGitIgnoreFilter(opts);
 
-	return globby(glob, {ignore: opts.ignores, nodir: true}).then(paths => {
+	return globby(patterns, {ignore: opts.ignores, nodir: true}).then(paths => {
 		// Filter out unwanted file extensions
 		// for silly users that don't specify an extension in the glob pattern
-		paths = paths.filter(x => {
-			// Remove dot before the actual extension
-			const ext = path.extname(x).replace('.', '');
-			return opts.extensions.indexOf(ext) !== -1;
-		});
+
+		paths = paths
+			.filter(ignoreFilter)
```

Comment:
Is there a good reason to filter out `.gitignore` patterns _after_ `globby`?

I believe this would be a major performance hit, e.g. if we conisder the `.gitignore` file below, then all files in `node_modules` will be globbed but filtered afterwards:

```
# .gitignore
node_modules
```